R/animint.R

```diff
@@ -1644,6 +1746,36 @@ getLegendList <- function(plistextra){
     geom.unique.rows <- sapply(geom.unique.list, nrow)
     is.ignored <- 1 < geom.data.rows & geom.unique.rows == 1
     gdefs[[leg]]$geoms <- geom.list[!is.ignored]
+    
+    ## Pass a geom.legend.list to be used by the
+    ## GetLegend function
+    geom.legend.list <- list()
+    for(geom.i in seq_along(gdefs[[leg]]$geoms)){
+      data.geom.i <- gdefs[[leg]]$geoms[[geom.i]]$data
+      params.geom.i <- gdefs[[leg]]$geoms[[geom.i]]$params
+      size.geom.i <- gdefs[[leg]]$geoms[[geom.i]]$size
+      
+      suppressWarnings(draw.key.used <- 
+                         gdefs[[leg]]$geoms[[geom.i]]$draw_key(
+                           data.geom.i, params.geom.i, size.geom.i)
+      )
+      geom.legend <- class(draw.key.used)[[1]]
+      geom.legend.list <- c(geom.legend.list, geom.legend)
+    }
+    
+    ## Process names to be used by the CleanData function
+    for(i in seq_along(geom.legend.list)){
+      if(geom.legend.list[[i]] %in% c("points")){
+        geom.legend.list[[i]] <- substr(geom.legend.list[[i]], 1,
+                                        nchar(geom.legend.list[[i]])-1)
+      }else if(geom.legend.list[[i]] %in% c("segments")){
+        geom.legend.list[[i]] <- "path"
+      }else if(geom.legend.list[[i]] %in% c("rect")){
+        geom.legend.list[[i]] <- "polygon"
+      }
+    }
+    
+    gdefs[[leg]]$geom.legend.list <- geom.legend.list
```

Comment:
the more R-ish way to do this would be using data and vector operations rather than a for loop, e.g.

```r
convert.names.vec <- c(segments="path", rect="polygon")
geom.legend.list[geom.legend.list %in% names(convert.names.vec)] <- convert.names.vec[geom.legend.list]
```

the advantage is that the mapping is defined in data so the code is clearer and easier to understand